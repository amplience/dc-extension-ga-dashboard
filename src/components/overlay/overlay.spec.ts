import { render, screen, fireEvent } from '@testing-library/svelte';
import Overlay from './overlay.svelte';

describe('Overlay', () => {
  it('should render an Overlay component', async () => {
    const { container } = render(Overlay, { onClick: jest.fn() });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an Overlay component with updated z-index', async () => {
    const { container } = render(Overlay, { onClick: jest.fn(), zIndex: 111 });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should trigger the onClick function', async () => {
    const mockOnClick = jest.fn();
    render(Overlay, { onClick: mockOnClick });

    const element = screen.getByTestId('overlay');

    await fireEvent.click(element);

    expect(mockOnClick).toHaveBeenCalled();
  });
});
