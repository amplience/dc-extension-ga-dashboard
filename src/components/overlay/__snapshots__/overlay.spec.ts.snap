// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render an Overlay component 1`] = `
<div>
  <div
    class="svelte-1k192f9"
    data-testid="overlay"
    style="z-index: 2"
  />
</div>
`;

exports[`Overlay should render an Overlay component with updated z-index 1`] = `
<div>
  <div
    class="svelte-1k192f9"
    data-testid="overlay"
    style="z-index: 111"
  />
</div>
`;
